pkg/ccl/changefeedccl/kvfeed/physical_kv_feed.go

```diff
@@ -179,9 +190,14 @@ func (p *rangefeed) addEventsToBuffer(ctx context.Context) error {
 				// expect SST ingestion into spans with active changefeeds.
 				return errors.Errorf("unexpected SST ingestion: %v", t)
 			case *kvpb.RangeFeedBulkEvents:
-				// Should be disabled so this is unreachable.
-				return errors.Errorf("unexpected bulk delivery: %v", t)
-
+				// TODO(#138346): We can handle these more gracefully once we
+				// migrate to the new rangefeed client. Until then, this is
+				// still an improvement over not using these.
+				for _, e := range t.Events {
```

Comment:
I think `e` is allowed to be any kind of event, not specifically just a value events. The approach in kvclient/rangefeed was to just but the whole switch over event type in the helper func and delegate to the switch for each (untyped) event in events.

Reply:
https://github.com/cockroachdb/cockroach/blob/msater/pkg/kv/kvclient/rangefeed/rangefeed.go#L522-L527